sync/src/synchronizer/block_fetcher.rs

```diff
@@ -151,6 +121,9 @@ impl<'a> BlockFetcher<'a> {
                 if status.contains(BlockStatus::BLOCK_STORED) {
                     // If the block is stored, its ancestor must on store
                     // So we can skip the search of this space directly
+                    self.synchronizer
+                        .peers()
+                        .set_last_common_header(self.peer, header.clone());
```

Comment:
It's better to also slide the window as well.


```suggestion
                    end = min(best_known.number(), header.number() + 1 + BLOCK_DOWNLOAD_WINDOW);
                    self.synchronizer
                        .peers()
                        .set_last_common_header(self.peer, header.clone());
```

This also requires change end to `let mut end`.

Reply:
1493a7773969bcf5f565ff8bf3c2c7bd6eaa085d merges the suggested change manually.